generate-db.js

```diff
@@ -20,57 +20,84 @@ const binPrefix = prefix.substring(1) + "/bin/";
-        res.on("data", (chunk) => {
-          rawData.push(Buffer.from(chunk, "binary"));
+    let reqUrl = `${repositoryURL}/${repo.name}/dists/${repo.distribution}/Contents-${arch}.gz`;
+    promises.push(fetchFile(reqUrl).then((rawBuffer) => {
+      let binMap = {};
+      let ungzipped = zlib.gunzipSync(rawBuffer).toString();
+      let linesContainingPathToBinaries = ungzipped
+        .split("\n")
+        .filter((line) => {
+          return line.startsWith(binPrefix);
         });
-        res.on("end", () => {
-          let binMap = {};
-          let rawBuffer = Buffer.concat(rawData);
-          let ungzipped = zlib.gunzipSync(rawBuffer).toString();
-          let linesContainingPathToBinaries = ungzipped
-            .split("\n")
-            .filter((line) => {
-              return line.startsWith(binPrefix);
-            });
-          linesContainingPathToBinaries.forEach((line) => {
-            const [pathToBinary, packageNames] = line.split(" ");
-            const binary = pathToBinary.substring(
-              pathToBinary.lastIndexOf("/") + 1
-            );
-            const packages = packageNames.split(",");
-
-            packages.forEach((packageName) => {
-              if (binMap[packageName] === undefined) {
-                binMap[packageName] = [];
-              }
-              binMap[packageName].push(binary);
-            });
-          });
-          const headerFile = `commands-${arch}-${repo.name}.h`;
-          if (fs.existsSync(headerFile)) {
-            fs.rmSync(headerFile);
+      linesContainingPathToBinaries.forEach((line) => {
+        const [pathToBinary, packageNames] = line.split(" ");
+        const binary = pathToBinary.substring(
+          pathToBinary.lastIndexOf("/") + 1
+        );
+        const packages = packageNames.split(",");
+        packages.forEach((packageName) => {
+          if (binMap[packageName] === undefined) {
+            binMap[packageName] = [];
           }
-          Object.keys(binMap)
-            .sort()
-            .forEach((packageName) => {
-              fs.appendFileSync(headerFile, `"${packageName}",\n`);
-              binMap[packageName].sort().forEach((packageName) => {
-                fs.appendFileSync(headerFile, `" ${packageName}",\n`);
-              });
-            });
+          binMap[packageName].push(binary);
+        });
+      });
+      const unjoinedContent = [];
+      Object.keys(binMap)
+        .sort()
+        .forEach((packageName) => {
+          unjoinedContent.push(`"${packageName}",\n`);
+          binMap[packageName].sort().forEach((packageName) => {
+            unjoinedContent.push(`" ${packageName}",\n`);
+          });
         });
+      const content = unjoinedContent.join("");
+
+      const headerFile = `commands-${arch}-${repo.name}.h`;
+      if (fs.existsSync(headerFile)) {
+        fs.rmSync(headerFile);
       }
-    );
+      const encoding = "utf-8";
+      fs.writeFileSync(headerFile, content, encoding);
```

Comment:
Currently my incorrect "fix" did nothing but creating empty files so that the compiler won't refuse to continue. However this obviously is not a proper fix at all.